### DISCUSSION TITLE
v1.5.1 release breaks usability

### DISCUSSION THREAD

**Discussion by l4rm4nd** [plausible/analytics#3052] [Self-Hosted Support]
### Past Issues Searched

- [X] I have searched open and closed issues to make sure that the bug has not yet been reported

### Issue is a Bug Report

- [X] This is a bug report and not a feature request, nor asking for self-hosted support

### Using official Plausible Cloud hosting or self-hosting?

Selhosted via Docker

### Describe the bug

Upon upgrading the plausible docker container to 1.5.1, the logs show the following errors:

````
Loading plausible..
Starting dependencies..
Starting repos..
create Plausible.Repo database if it doesn't exist
create Plausible.ClickhouseRepo database if it doesn't exist
Creation of Db successful!
Loading plausible..
Starting dependencies..
Starting repos..
Running migrations for Elixir.Plausible.Repo
Running migrations for Elixir.Plausible.ClickhouseRepo
Migrations successful!
init-admin is deprecated and is no-op now
user registration on first launch happens via Web UI instead
07:03:16.373 [error] #PID<0.3216.0> running PlausibleWeb.Endpoint (connection #PID<0.3200.0>, stream id 16) terminated
Server: analytics.example.com:80 (http)
Request: GET /blog.example.com
** (exit) an exception was raised:
    ** (Postgrex.Error) ERROR 42703 (undefined_column) column g0.domain does not exist
    query: SELECT TRUE FROM "goals" AS g0 WHERE (g0."domain" = $1) LIMIT 1
        (ecto_sql 3.8.3) lib/ecto/adapters/sql.ex:932: Ecto.Adapters.SQL.raise_sql_call_error/1
        (ecto_sql 3.8.3) lib/ecto/adapters/sql.ex:847: Ecto.Adapters.SQL.execute/6
        (ecto 3.8.4) lib/ecto/repo/queryable.ex:221: Ecto.Repo.Queryable.execute/4
        (ecto 3.8.4) lib/ecto/repo/queryable.ex:19: Ecto.Repo.Queryable.all/3
        (ecto 3.8.4) lib/ecto/repo/queryable.ex:130: Ecto.Repo.Queryable.exists?/3
        (plausible 0.0.1) lib/plausible_web/controllers/stats_controller.ex:66: PlausibleWeb.StatsController.stats/2
        (plausible 0.0.1) lib/plausible_web/controllers/stats_controller.ex:1: PlausibleWeb.StatsController.action/2
        (plausible 0.0.1) lib/plausible_web/controllers/stats_controller.ex:1: PlausibleWeb.StatsController.phoenix_controller_pipeline/2
07:03:21.327 [error] #PID<0.3219.0> running PlausibleWeb.Endpoint (connection #PID<0.3200.0>, stream id 19) terminated
Server: analytics.example.com:80 (http)
````

The postgresql db container shows the following:
````
2023-06-15 04:03:03.765 UTC [22] LOG:  redo is not required
2023-06-15 04:03:03.832 UTC [1] LOG:  database system is ready to accept connections
2023-06-19 06:58:53.155 UTC [11564] ERROR:  column g0.domain does not exist at character 39
2023-06-19 06:58:53.155 UTC [11564] STATEMENT:  SELECT TRUE FROM "goals" AS g0 WHERE (g0."domain" = $1) LIMIT 1
````

The web application shows the following error:
![image](https://github.com/plausible/analytics/assets/21357789/b2b73a69-fe88-4279-8894-5cc5f6f45eef)
****

### Expected behavior

It should be possible to inspect site statistics via the plausible web application. 

### Environment

```markdown
- OS: Windows 11
- Browser: Microsoft Edge
```

**Comment by ruslandoga** [comment]
👋 @l4rm4nd 

Please check out https://github.com/plausible/analytics/discussions/3047#discussioncomment-6215056

**Reply by ruslandoga** [reply]
And if it doesn't help, please post your `docker-compose.yml` and any other self-hosting info that you consider relevant :)

**Reply by l4rm4nd** [reply]
Hi @ruslandoga,

many thanks for the linked issue. I've identified the same flaw of accidentially upgrading to v2.0.0 first and then reverting back. This bricks the postgres db as mentioned.

Currently pulling my backups from yesterday. Should be fine again in a few minutes.

Will use the v1.5 tag specifically and not use latest again. My mistake, sorry!

**Reply by ruslandoga** [reply]
I'm glad you had backups!

> My mistake, sorry!

You are too kind. The mistake is completely mine and it's me who should be sorry.
And I am!
Sorry!

**Reply by ruslandoga** [reply]
Just a note, `v2.0.0-rc.2` would be pretty much the same as `v2.0.0`, and `v2.0.0` will probably be released this week. 

The only difference between `rc.2` and final is that some feature flags would have been removed from `v2.0.0` (like the ones that are hiding some extra visitor stats in UI). So it should be safe to upgrade to `v2.0.0-rc.2` without the need for restoring from backups.

**Reply by l4rm4nd** [reply]
As said, I pulled the postgresql backups and I am back online again.

Problem solved. Stupid rookie mistake of using a latest tag.

Thanks for your help and the availability of plausible for us all in general!
